pandas/core/indexes/api.py

```diff
@@ -58,6 +59,7 @@
     "get_objs_combined_axis",
     "union_indexes",
     "get_consensus_names",
+    "get_unanimous_names",
```

Comment:
can we totally remove get_consenus_names? (alt could cal it `get_result_name` and use these routines under the hood)

Reply:
Only other place `get_consensus_names` is used is in `concat` for the nested MultiIndex case. In fact, looking at master, this ends up dropping level names even when they all agree since it operates jointly on all levels, rather than going level-by-level like `get_unanimous_names`. Swapping in `get_unanimous_names` here lets us totally drop `get_consensus_names` and seems to work very well, all tests passed.